packages/cli/src/rollup-plugins/typescript-declarations/common.ts

```diff
@@ -195,11 +195,32 @@ export const getDeclarationsForFile = async (
         (context) => (
           node
         ): import("typescript").Bundle | import("typescript").SourceFile => {
-          if (!visitModuleSpecifier) {
-            return node;
-          }
-
-          const cachedVisitModuleSpecifier = memoize(visitModuleSpecifier);
+          const replaceTsExtensions = (moduleSpecifier: string) => {
+            if (!moduleSpecifier.startsWith(".")) {
+              return moduleSpecifier;
+            }
+            return moduleSpecifier.replace(
+              /(\.d)?\.(ts|tsx|mts|cts)$/,
+              (match) => {
```

Comment:
I also decided to replace those with runtime equivalents because that works in older TS versions and it seems to be more compatible with the stricter rules imposed by `type:module` and stuff. While we don't really support `type:module` yet, we will at some point. So to reduce conditional logic for the future, I just went with replacing - instead of stripping those away.